spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/source/SparkScan.java

```diff
@@ -175,7 +181,25 @@ public Statistics estimateStatistics() {
   protected Statistics estimateStatistics(Snapshot snapshot) {
     // its a fresh table, no data
     if (snapshot == null) {
-      return new Stats(0L, 0L);
+      return new Stats(0L, 0L, Maps.newHashMap());
+    }
+
+    Map<NamedReference, ColumnStatistics> map = Maps.newHashMap();
```

Comment:
Can we have a more descriptive name?

Reply:
Also can we avoid the allocation if enableColumnStats and CBO aren't enabled?

Reply:
Changed to `colStatsMap` and also only allocate if enableColumnStats and CBO are enabled.